File: day-01/part-2.ts
import { loadInput } from "../utils";


const sum = (arr: number[]) => arr.reduce((acc, el) => acc + el, 0);

const input = loadInput("input");
const elfs = input.split("\n\n");

const cals = elfs.map(cals => cals.split("\n").map(cal => parseInt(cal, 10)));
const calsPerElf = cals.map(sum);

calsPerElf.sort((a, b) => b - a);
const [a, b, c] = calsPerElf;

console.log(a + b + c);

File: kotlin/src/main/kotlin/day01/part-2.kt
package day01.part2

import loadInput

fun main() {
    val input = loadInput("./src/main/kotlin/day01/input")

    val elfs = input.split("\n\n")
    val cals = elfs.map { it.split("\n").map { it.toInt() } }

    val total = cals.map { it -> it.sum() }
    val top3 = total.sortedDescending().take(3)
    println(top3.sum())
}
